packages/components/top-menu/top-menu.ts

```diff
@@ -0,0 +1,59 @@
+import {
+    booleanAttribute,
+    ChangeDetectionStrategy,
+    Component,
+    Directive,
+    input,
+    Input,
+    ViewEncapsulation
+} from '@angular/core';
+
+@Directive({
+    standalone: true,
+    selector: '[kbq-top-menu-spacer]',
+    host: {
+        class: 'kbq-top-menu-spacer'
+    }
+})
+export class KbqTopMenuSpacer {}
+
+/**
+ * Directive that dynamically applying CSS classes based on a placement value (left or right).
+ */
+@Directive({
+    standalone: true,
+    selector: '[kbq-top-menu-container]',
+    host: {
+        class: 'kbq-top-menu-container',
+        '[class.kbq-top-menu-container__left]': 'placement() === "left"',
+        '[class.kbq-top-menu-container__right]': 'placement() === "right"'
+    }
+})
+export class KbqTopMenuContainer {
+    // Conditionally applies a CSS class based on the value
+    placement = input.required<'left' | 'right'>();
+}
+
+@Component({
+    standalone: true,
+    selector: 'kbq-top-menu',
+    template: `
+        <ng-content />
+    `,
+    styleUrls: [
+        './top-menu.scss',
+        './top-menu-tokens.scss'
+    ],
+    host: {
+        class: 'kbq-top-menu',
+        '[class.kbq-top-menu-overflow]': 'hasOverflow'
+    },
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None
+})
+export class KbqTopMenu {
+    /**
+     * Enables overflow behavior, applying `kbq-top-menu-overflow` to show a bottom shadow.
+     */
+    @Input({ transform: booleanAttribute }) hasOverflow: boolean = false;
```

Comment:
```suggestion
    @Input({ transform: booleanAttribute }) withShadow: boolean = false;
```
мб инпут и селектор так и назвать? из одного названия переменной понятно, за что она отвечает, что думаешь?

Reply:
Да, согласен